### PR TITLE
Upgrade Ray to 2.2.x and PyArrow to 7+

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
### Detail
Upgrading PyArrow to 7+ so that we can merge the DeltaLake feature into `release-3.0.0`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: f06ae6f30c6d4dda8afb0ce30523557eb7000747
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=624f90f5-beab-47d5-8503-8686f5fb2477%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: f06ae6f30c6d4dda8afb0ce30523557eb7000747
* Result: FAILED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=56d81654-cb51-4b36-b010-e8b86a5d449b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: f06ae6f30c6d4dda8afb0ce30523557eb7000747
* Result: SUCCEEDED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ea8ffeaf-2030-4486-96f0-bf6f1babb99b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 7116645f07623274b4fde1ca78cae2cac8e812a6
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=16244af5-c1ae-49af-9a8c-f74baff3e83d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: 7116645f07623274b4fde1ca78cae2cac8e812a6
* Result: FAILED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7a6739b6-e6f1-4e0c-a89c-fd3be2aec577%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 7116645f07623274b4fde1ca78cae2cac8e812a6
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1eb19071-4194-4c81-b279-0d4c83d786d3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: a647da299d2bd94d2b9670e873e93bc62b68f09b
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9a2bf06b-1b08-4af8-81d6-83c85e1d87b0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: a647da299d2bd94d2b9670e873e93bc62b68f09b
* Result: FAILED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=79b00fa2-5ea2-47c0-9b45-9946a5434cd5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: a647da299d2bd94d2b9670e873e93bc62b68f09b
* Result: SUCCEEDED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=af2dc02d-5f84-448d-8304-daf8c6fb04bc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: pyproject.toml [hunk 1]
```diff
@@ -47,7 +47,7 @@ backoff = ">=2.0.0,<3.0.0"
 SPARQLWrapper = { version = "^2.0.0", optional = true }
 pyodbc = { version = "~4.0.32", optional = true }
 modin = { version = "~0.16.0", optional = true }
-ray = { version = "~2.0.0", extras = ["default", "data"], optional = true}
+ray = { version = "^2.2.0", extras = ["default", "data"], optional = true}
```
Comment (jaidisido) [inline pyproject.toml:50]: Hmm the only risk here is that the Glue team does not upgrade as well
What is the risk of allowing `>=2.0, <2.3.0`?

File: tests/unit/test_s3_parquet.py [hunk 2]
```diff
@@ -524,6 +524,7 @@ def test_timezone_raw_values(path):
 
 
 @pytest.mark.parametrize("partition_cols", [None, ["a"], ["a", "b"]])
+@pytest.mark.xfail(reason="Empty file cannot be read by Ray", raises=AssertionError, condition=is_ray_modin)
```
Comment (jaidisido) [inline tests/unit/test_s3_parquet.py:527]: Would you know why it suddenly started failing following upgrade?

File: pyproject.toml (hunk 1 above)
Comment (LeonLuttenberger) [reply pyproject.toml:50]: We want at least `2.2` so that we can get PyArrow 7+. If we don't do that, we'll have a lot of difficulties merging the `deltalake` feature.

In terms of `<2.3.0`, do we have a good reason to block future minor versions of Ray?

File: tests/unit/test_s3_parquet.py (hunk 2 above)
Comment (LeonLuttenberger) [reply tests/unit/test_s3_parquet.py:527]: I'm not sure, especially as it's only failing in distributed mode, which leads me to think it's Ray, not PyArrow.

It seems like a rare enough edge case though where I feel like it's more important to make the upgrade, and potentially work out this kink later. But let me know if you disagree.

File: pyproject.toml (hunk 1 above)
Comment (jaidisido) [reply pyproject.toml:50]: But I would have thought that even with ray 2.0 to 2.2, if someone tries to install delta lake 6+, the pip resolver would realize that it needs to choose ray 2.2 to enable pyarrow 7 no?

Comment (jaidisido) [reply pyproject.toml:50]: I think we import some private ray modules and we are worried that an upgrade would break that if they were to modify them

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: 991582b18cece1737d6a30f5ef9f257c35e9ed1c
* Result: SUCCEEDED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1ead7800-aa2d-49ea-96d4-b717ab438cc8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 991582b18cece1737d6a30f5ef9f257c35e9ed1c
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cc3a420c-d9f9-469e-ab01-f9528a5feb74%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by jaidisido — APPROVED** [review]
